rust/core-lib/src/editor.rs

```diff
@@ -386,19 +398,25 @@ impl<W: Write + Send + 'static> Editor<W> {
     }
 
     fn delete(&mut self) {
-        let start = if self.view.sel_start != self.view.sel_end {
-            self.view.sel_min()
-        } else if let Some(bsp_pos) = self.text.prev_codepoint_offset(self.view.sel_end) {
-            // TODO: implement complex emoji logic
-            bsp_pos
-        } else {
-            self.view.sel_max()
-        };
+        let mut builder = delta::Builder::new(self.text.len());
+        for region in self.view.sel_regions() {
+            let start = if !region.is_caret() {
+                region.min()
+            } else if let Some(bsp_pos) = self.text.prev_codepoint_offset(region.end) {
+                // TODO: implement complex emoji logic
+                bsp_pos
+            } else {
+                region.end
```

Comment:
`unwrap_or` is cleaner, will do that. Yes, that handles beginning of buffer, so it is reachable.